crates/orchestrator/src/jobs/da_job/mod.rs

```diff
@@ -394,6 +395,30 @@ fn da_word(class_flag: bool, nonce_change: Option<Felt>, num_changes: u64) -> Fe
     Felt::from_dec_str(&decimal_string).expect("issue while converting to fieldElement")
 }
 
+fn refactor_state_update(state_update: &mut StateDiff) {
+    let addresses_in_nonces: Vec<Felt> = state_update.nonces.clone().iter().map(|item| item.contract_address).collect();
+    let addresses_in_storage_diffs: Vec<Felt> =
+        state_update.storage_diffs.clone().iter().map(|item| item.address).collect();
+
+    let address_to_insert = find_unique_addresses(addresses_in_nonces, addresses_in_storage_diffs);
+
+    for address in address_to_insert {
+        state_update.storage_diffs.push(ContractStorageDiffItem { address, storage_entries: vec![] })
+    }
+}
+
+fn find_unique_addresses(nonce_addresses: Vec<Felt>, storage_diff_addresses: Vec<Felt>) -> Vec<Felt> {
```

Comment:
This seems to not be the best optimised code,
This has a time complexity of O(n*m), 
I think we can simply achieve O(n+m) by converting storage_diff_addresses to a hashset.



```rust
fn find_unique_addresses(nonce_addresses: Vec<Felt>, storage_diff_addresses: Vec<Felt>) -> Vec<Felt> {
    let storage_set: HashSet<_> = storage_diff_addresses.into_iter().collect();
   
    nonce_addresses
        .into_iter()
        .filter(|addr| !storage_set.contains(addr))
        .collect()
}
```

Reply:
nice fixed